Code/Framework/AzFramework/Tests/Platform/Common/Xcb/Printers.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <ostream>
+#include <AzFramework/Input/Channels/InputChannel.h>
+
+namespace AzFramework
+{
+    inline void PrintTo(const InputChannel::State& state, std::ostream* os)
+    {
+        switch(state)
+        {
+            case InputChannel::State::Began:
+                *os << "Began";
+                break;
+            case InputChannel::State::Ended:
+                *os << "Ended";
+                break;
+            case InputChannel::State::Idle:
+                *os << "Idle";
+                break;
+            case InputChannel::State::Updated:
+                *os << "Updated";
+                break;
+        }
+    }
+} // namespace AzFramework
```

Comment:
For compilation performance, I think it would be better to moves this implementation to a Printers.cpp file.

That way neither `ostream` nor `InputChannel.h` needs to be included.

The `InputChannel::State` enum and `std::ostream` can be forward declared (assuming InputChannel is a namespace and not a class)

Reply:
I still have to include `iosfwd` to forward-declare `std::ostream`. And `AzFramework::InputChannel` is a class that defines a nested scoped enum:
```
namespace AzFramework
{
    class InputChannel : public InputChannelRequestBus::Handler
    {
        enum class State
        {
            Idle,    //!< Examples: inactive or idle, not currently emitting input events
            Began,   //!< Examples: button pressed, trigger engaged, thumb-stick exits deadzone
            Updated, //!< Examples: button held, trigger changed, thumb-stick is outside deadzone
            Ended    //!< Examples: button released, trigger released, thumb-stick enters deadzone
        };
```
I don't think I can forward declare that enum. I _can_ move the implementation out of the header though, I'm just not sure how much benefit it'll bring.

Reply:
Well it would provide a pattern for others working on Xcb system to follow, if we need to add more googletest printers.

I know @nemerle has been doing a lot of work to reduce our include overhead, so I think we should try to follow that pattern.